RIOT-OS/RIOT
# tests/gnrc_netif_delayed_start: add test for delayed start of interfaces

### Contribution description

This tests application defined startup & shutdown of network interfaces.

Battery powered devices might only want to enable the radio if they have data to send.
For the most power saving and least startup time, that means they will not even configure the radio when not needed and just leave it in Deep Sleep.

For this to work the radio must remain in deep sleep across reboots and also allow to be shut down properly.

This adds a test to exercise that use-case.


### Testing procedure

Attach an ammeter to the supply line of your networking device.
 - run the `start` command to bring up your network interfaces. Current draw should be at `IDLE`/`RX` level.
  - run the `shutdown` command to put your network interfaces to sleep. Current draw should be at the lowest `SLEEP` level.
 - run the `reboot` command to confirm your interfaces remain sleeping across reboots

### Issues/PRs references
